src/docs/grains/grain_persistence/relational_storage.md

```diff
@@ -93,122 +93,121 @@ public class AdoNetGrainStorageOptions
+> When a table is small, it almost doesn’t matter what the query plan is. When it’s medium, an OK query plan is fine, but when it's huge (millions upon millions or billions of rows), even a slight variation in the query plan can kill you. For this reason, we hint our sensitive queries heavily.
 
-### 5. No assumptions on what tools, libraries or deployment processes are used in organizations
+### 5. No assumptions on what tools, libraries, or deployment processes are used in organizations
 
 Many organizations have familiarity with a certain set of database tools, examples being [Dacpac](https://docs.microsoft.com/en-us/sql/relational-databases/data-tier-applications/data-tier-applications)
-or [Red Gate](https://www.red-gate.com/). It may be so that deploying a database requires either a permission or a person, such as someone
+or [Red Gate](https://www.red-gate.com/). It may be that deploying a database requires either a permission or a person, such as someone
 in a DBA role, to do it. Usually this means also having the target database layout and a rough sketch of the queries the application will
-produce to the database to be used estimate the load. There might be processes, perhaps influenced by industry standards, which mandate script based deployment.
+produce for use in estimating the load. There might be processes, perhaps influenced by industry standards, which mandate script-based deployment.
 Having the queries and database structures in an external script makes this possible.
 
-### 6. Use the minimum set needed of interface functionality to load the ADO.NET libraries and functionality
+### 6. Use the minimum set of interface functionality needed to load the ADO.NET libraries and functionality
 
-This is both fast and has less surface to be exposed to ADO.NET library implementation discrepancies.
+This is both fast, and has less surface exposed to the ADO.NET library implementation discrepancies.
 
 ### 7. Make the design shardable
 
-When it makes sense, for instance in relational storage provider, make the design readily shardable. This means for instance no database dependent
-data (e.g. `IDENTITY`) and basically it means the information that distinguishes row data should build on only data from the actual parameters.
+When it makes sense, for instance in a relational storage provider, make the design readily shardable. For instance, this means using no database-dependent
+data (e.g. `IDENTITY`). Information that distinguishes row data should build on only data from the actual parameters.
 
 ### 8. Make the design easy to test
 
-Creating a new backend should be ideally as easy as translating one of the deployment scripts and adding a new connection string to tests assuming default
-parameters, check if a given database is installed and then run the tests against it.
+Creating a new backend should ideally be as easy as translating one of the deployment scripts, adding a new connection string to tests (assuming default
```

Comment:
@voipengineer "translating" here could be elaborated into "translating one of the existing deployment scripts into the SQL dialect of the backend you are trying to target" - @veikkoeeva, how does that sound to you? Maybe I have it wrong.

Reply:
@ReubenBond, @voipengineer This looks understanable to me.

I note also on

> The appropriate ADO.NET library must be loaded into the process. This should be defined as usual, e.g. via the [DbProviderFactories](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/obtaining-a-dbproviderfactory) element in the application configuration.

This is not done currently. These were briefly removed from .NET Standard, so a custom loader was introduced that did mostly the same. The codebase currently does not use this type, but should be refactored to make use of it again.

As for context: some internal things are to go around the era when only connection string came through Orleans system and it was fixed to SQL Server. This is a note that does not need to show in the text, I'll just write here for the interested.

Reply:
I have incorporated @ReubenBond 's suggestions. I don't know if @veikkoeeva 's comments were meant to be incorporated, or where, so I have not attempted to do so.